src/include/stir/DetectorCoordinateMap.h

```diff
@@ -65,14 +66,32 @@ class DetectorCoordinateMap
 
 	//! Constructor calls read_detectormap_from_file( filename ).
 	DetectorCoordinateMap(const std::string& filename, double sigma = 0.0) :
-		sigma(sigma),
-		distribution(0.0, sigma)
-		{ read_detectormap_from_file( filename ); }
+		sigma(sigma)
+		{ 
+      read_detectormap_from_file(filename);
+#ifdef STIR_OPENMP
+      distributions.resize(omp_get_max_threads());
+      for (auto &distribution : distributions)
+        distribution = std::normal_distribution<double>(0.0, sigma);
+#else
+      distributions.resize(1);
+      distributions[0] = std::normal_distribution<double>(0.0, sigma);
+#endif
```

Comment:
```suggestion
		DetectorCoordinateMap(sigma)
		{ 
                   read_detectormap_from_file(filename);
```
should work I believe (delegating constructor)

Reply:
True, that's much cleaner!